cf-check/observables.c

```diff
@@ -0,0 +1,103 @@
+#include <platform.h>
+#include <logging.h>
+#include <definitions.h> // CF_BUFSIZE
+#include <known_dirs.h> // GetStateDir()
+#include <file_lib.h> // FILE_SEPARATOR
+#include <alloc.h> // xstrdup()
+#include <db_structs.h> // observable_strings
+#include <string_lib.h> // StringEqual
+#include <observables.h>
+
+/**
+ * GetObservableNames guarantees to return an array of CF_OBSERVABLES size with non-NULL strings
+ * if using built-in, will generate "observable[n]" name, if reading ts_key file and the item
+ * is "spare" then generating "spare[n]" where 'n' is the id number which could be helpful
+ * in debugging raw observables data.
+ *
+ * This function is similar to Nova_LoadSlots() in libpromises/monitoring_read.c
+ * If we refactor for dynamic observables instead of hard coded and limited to 100
+ * then we likely should change here and there or refactor to have
+ * this ts_key read/parse code in one shared place.
+ */
+void GetObservableNames(const char *ts_key_path, char ***obnames)
```

Comment:
It might make more sense to return `char **` here instead of the _out_ parameter and `void` as return type.

Reply:
agreed. done.